plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -14,14 +14,61 @@
  * limitations under the License.
  */
 
-import React, { ChangeEvent, useEffect, useState } from 'react';
+import React, { ChangeEvent, useEffect, KeyboardEvent, useState } from 'react';
 import { useDebounce } from 'react-use';
 import { InputBase, InputAdornment, IconButton } from '@material-ui/core';
 import SearchIcon from '@material-ui/icons/Search';
 import ClearButton from '@material-ui/icons/Clear';
 
 import { useSearch } from '../SearchContext';
 
+type PresenterProps = {
+  value: string;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  handleClear?: () => void;
+  onKeyDown?: (e: KeyboardEvent<HTMLInputElement>) => void;
```

Comment:
is it cleaner to expose onSubmit instead? we could react both to click-button presses and enter presses that way and show the surrounding component what happened in a coherent way

Reply:
Yeah the more I think of it, only onChange + onSubmit as the entire interface would be the API a consumer wanted to see, no clears or keydowns

Reply:
Alright! My idea here was that the presenter should be unaware of the logic, and if we expose `onSubmit` instead of `onKeyDown`, then it has to implement the logic for submitting on its own, and it makes the API less powerful, since the user can't access the `keyDown` event. However, I think this is an acceptable trade-off, and `onSubmit` feels like a **way** more convenient prop.

I'm not sure I understand what you mean about "no clears" though. I don't see another way of supporting the clearing of the input field. Please clarify if I misunderstand!

Fixed in 259f741.